plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/embeddings/expression/OperatorExpEmbedding.kt

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2010-2024 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.formver.embeddings.expression
+
+import org.jetbrains.kotlin.formver.asPosition
+import org.jetbrains.kotlin.formver.domains.InjectionImageFunction
+import org.jetbrains.kotlin.formver.embeddings.SourceRole
+import org.jetbrains.kotlin.formver.embeddings.asInfo
+import org.jetbrains.kotlin.formver.embeddings.types.TypeEmbedding
+import org.jetbrains.kotlin.formver.linearization.LinearizationContext
+import org.jetbrains.kotlin.formver.viper.ast.Exp
+
+interface InjectionBasedExpEmbedding : DirectResultExpEmbedding {
```

Comment:
Could you elaborate on the sense in which it is `InjectionBased`?  How should people understand the members?